app/views/CreateDiscussionView/utils/handleLoadingChange.ts

```diff
@@ -0,0 +1,33 @@
+import I18n from '../../../i18n';
+import { sendLoadingEvent } from '../../../containers/Loading';
+import { getRoomTitle, showErrorAlert } from '../../../lib/methods/helpers';
+import { goRoom } from '../../../lib/methods/helpers/goRoom';
+import { IError, IResult } from '../interfaces';
+
+interface IHandleLoadingChange {
+	loading: boolean;
+	failure: boolean;
+	isMasterDetail: boolean;
+	error: IError;
+	result: IResult;
+}
+const handleLoadingChange = ({ loading, failure, isMasterDetail, error, result }: IHandleLoadingChange) => {
```

Comment:
I think it's misguiding to call it `handleLoadingChange`.
It's technically correct, of course, but what's actually happening here is we wait for the result of the submit to do something.

Loading gives the impression of being just a spinning indicator, when it's actually the main goal of the component 😅 

```suggestion
const handleSubmit = ({ loading, failure, isMasterDetail, error, result }: IHandleLoadingChange) => {
```

Could also be `handleSubmitResult` or `handleSubmitEvent`. It's up to you.